source/simulator/helper_functions.cc

```diff
@@ -985,49 +985,144 @@ namespace aspect
   template <int dim>
   void Simulator<dim>::apply_limiter_to_dg_solutions (const AdvectionField &advection_field)
   {
-    AssertThrow (dim == 2,
-                 ExcMessage ("The bound preserving limiter is currently only implemented for 2-dimensional problem."));
+    /*
+     * first setup the quadrature points which are used to find the maximum and minimum solution values at those points.
+     * To construct those special quadrature points:
+     * use combination of 1) one dimentional Gauss points; 2) one dimentional Gauss-Lobatto points;
```

Comment:
instead of "combination", can you say something like "a quadrature formula that combines all quadrature points constructed as all tensor products of 1) ...; 2) ..."